### ISSUE TITLE
transformers.image_transforms.normalize documents and checks for the wrong type for std and mean arguments

### ISSUE THREAD

**Issue by CalOmnie** [huggingface/transformers#35772]
### System Info

- `transformers` version: 4.49.0.dev0
- Platform: Linux-6.8.0-51-generic-x86_64-with-glibc2.39
- Python version: 3.11.11
- Huggingface_hub version: 0.27.1
- Safetensors version: 0.5.1
- Accelerate version: 1.2.1
- Accelerate config:    not found
- PyTorch version (GPU?): 2.5.1+cu124 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: Neither
- Using GPU in script?: No
- GPU type: NVIDIA GeForce GTX 1060 6GB

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
import numpy as np
from transformers.image_transforms import normalize

mock_img = np.random.rand(3, 10, 10)

# Simulating getting the std and mean from text
std = map(float, "1.5 0. 2.".split())
mean = map(float, "1.5 0. 2.".split())
# According to the doc, this setup should work, but it does not since "map" does not implement `len`

normalize(mock_img, std, mean)
```

### Expected behavior

Use the `Sequence` type instead of Iterable to let users know that `len` needs to be implemented.

**Comment by CalOmnie** [comment]
Question: Am I following the right process to submit very small fix that do not impact the runtime (e.g. typing and documentations)? I notices a few other mistypes while reading the code and I'd like to do it properly going forward.

**Comment by Rocketknight1** [comment]
Hi @CalOmnie, yes, this was good! For small fixes when the issue is clear, like bugs or obviously incorrect types, you can just submit a PR. We recommend opening an issue first when the change might be controversial or need discussion (for example, adding a whole new model class, or changing the behaviour of existing tools)
